solidity-its/src/main/java/org/sonarsource/solidity/its/SolidityRuling.java

```diff
@@ -43,17 +56,78 @@ public static void collectSolidityFiles() {
     Arrays.asList(getProjects())
       .stream()
       .forEach(projectName -> {
+        List<File> listFiles = new ArrayList<>();
         try {
           String path = String.format("%s%s", DIR, projectName);
           Files.find(Paths.get(path),
             Integer.MAX_VALUE,
             (filePath, fileAttr) -> fileAttr.isRegularFile())
             .filter(file -> file.getFileName().toString().endsWith(".sol"))
-            .forEach(file -> analyzeFile(file.toFile(), projectName));
+            .forEach(file -> listFiles.add(file.toFile()));
+          sortList(listFiles);
+          filesToAnalyze.put(projectName, listFiles);
+        } catch (IOException e) {
+          LOG.debug(e.getMessage(), e);
+        }
+      });
+  }
+
+  private static void sortList(List<File> listFiles) {
+    Collections.sort(listFiles, (File f1, File f2) -> f1.getName().compareTo(f2.getName()));
+  }
+
+  public static void collectFilesWithRecordedIssues() {
+
+    Arrays.asList(getProjects())
+      .stream()
+      .forEach(projectName -> {
+        String path = String.format("%s%s", SolidityRulingIts.RECORD_ISSUES, projectName);
+        try {
+          Files.find(Paths.get(path),
+            Integer.MAX_VALUE,
+            (filePath, fileAttr) -> fileAttr.isRegularFile())
+            .forEach(recordedIssues -> {
+              String actualIssuePath = String.format("%s%s", SolidityRulingIts.ACTUAL_ISSUES,
+                recordedIssues.toString().replaceAll(SolidityRulingIts.RECORD_ISSUES, ""));
+              try {
+                if (!FileUtils.contentEquals(new File(recordedIssues.toString()), new File(actualIssuePath))) {
+                  System.out.println("Not equal!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\nAAAAAAAAAAAAAAAAAAAAAAAAAAA     " + recordedIssues.toString() + " - " +
+                    actualIssuePath);
+                  System.out.println("------" + recordedIssues + "-------------");
+                  File file1 = new File(recordedIssues.toString());
+                  FileInputStream fis = new FileInputStream(file1);
+                  int oneByte;
+                  while ((oneByte = fis.read()) != -1) {
+                    System.out.print((char) oneByte);
+                  }
+                  System.out.println("------" + actualIssuePath + "-------------");
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Replace this use of System.out or System.err by a logger. ([squid:S106](https://sonarcloud.io/organizations/stylianos-agapiou-sonarsource-gi/rules#rule_key=squid%3AS106))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=org.sonarsource.solidity%3Asonar-solidity&pullRequest=31&issues=AWRqjT0tMt9UOhrwBY2d&open=AWRqjT0tMt9UOhrwBY2d)

[ISSUE_KEY]: AWRqjT0tMt9UOhrwBY2d